src/runtime/random.py

```diff
@@ -2,44 +2,58 @@
 
 e: f64 = 2.718281828459045235360287471352662497757
 eps: f64 = 1e-16
+_seed_set: bool = False
 
 #: TODO: Call `log` from C directly until we fix the multiple import issue
+
+
 def _log(x: f64) -> f64:
     return _lfortran_dlog(x)
 
+
 @ccall
 def _lfortran_dlog(x: f64) -> f64:
     pass
 
+
 def _exp(x: f64) -> f64:
     return e**x
 
+
 def _sqrt(x: f64) -> f64:
     return x**(1/2)
 
+
 def _abs(x: f64) -> f64:
     if x < 0.0:
         return -x
     return x
 
+
 def random() -> f64:
     """
     Returns a random floating point number in the range [0.0, 1.0)
     """
+    if not _seed_set:
+        _lfortran_init_random_clock()
```

Comment:
That's one way to do it, but a better way is to call the `init_random` at the beginning of the program, not at every call. We have infrastructure for this, I think there is a function that is being called at the beginning, so we just need to call this from it. 

Initially we can call it every time, later on we can update it to only call it if `random` is used.

Reply:
I cannot seem to find this function. Where is it, and where is it being called from?